exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java

```diff
@@ -143,6 +144,11 @@ public static final RelOptRule getDirFilterOnScan(OptimizerRulesContext optimize
   }
 
   protected void doOnMatch(RelOptRuleCall call, Filter filterRel, Project projectRel, TableScan scanRel) {
+    if (wasAllPartitionsPruned) {
```

Comment:
I feel this flag may lead to scenarios where pruning does not happen when it should.  The flag is PruneScanRule's internal variable. The same rule could be applied to pruning for multiple tables.  So, if PruneScanRule turns on this flag for T1, then we might skip pruning logic for T2 later on, when this rule is fired.